src/main/java/io/infracloud/cassandra/tracing/JaegerTracing.java

```diff
@@ -17,195 +17,201 @@
  */
 package io.infracloud.cassandra.tracing;
 
-import com.google.common.collect.ImmutableMap;
 import io.jaegertracing.Configuration;
-import io.jaegertracing.internal.JaegerTracer;
-import io.jaegertracing.internal.JaegerSpanContext;
 import io.jaegertracing.internal.JaegerSpan;
-import io.opentracing.Tracer;
+import io.jaegertracing.internal.JaegerSpanContext;
+import io.jaegertracing.internal.JaegerTracer;
+import io.jaegertracing.internal.propagation.TextMapCodec;
+import io.opentracing.SpanContext;
 import io.opentracing.propagation.Format;
-import io.opentracing.propagation.TextMap;
 import org.apache.cassandra.config.DatabaseDescriptor;
 import org.apache.cassandra.net.MessageIn;
 import org.apache.cassandra.tracing.TraceState;
 import org.apache.cassandra.tracing.Tracing;
 import org.apache.cassandra.utils.FBUtilities;
 import org.apache.cassandra.utils.UUIDGen;
-import org.slf4j.Logger;
-import org.slf4j.LoggerFactory;
 
 import java.net.InetAddress;
 import java.nio.ByteBuffer;
 import java.time.Instant;
+import java.util.HashMap;
 import java.util.Map;
 import java.util.UUID;
 
-public final class JaegerTracing extends Tracing
-{
-    // the key mentioned here will be used when sending the call to
-    // Cassandra i.e. with customPayload
+/**
+ * A single instance of this is created for entire Cassandra, so we need to use thread-local
+ * storage.
+ * <p>
+ * Now, there are two possibilities. Either we are the coordinator, in which case Cassandra will call:
+ * 1. newSession(...) by Native-Transport-Request
+ * 2.   which is called by (1). newTraceState(...)
+ * 3. begin(...)
+ * 4. trace(...)
+ * 5. stopSessionImpl(...)
```

Comment:
Is the formatting correct here? I'm not able to understand the point 2 correctly.

Reply:
Yes, it's indented on purpose.